enhancements/network/dual-stack.md

```diff
@@ -0,0 +1,236 @@
+the `IPv6DualStack` feature gate in `kubelet`, `kube-apiserver`, and
+`kube-controller-manager`. We tried to do this for all clusters in 4.4
+but [ran into bugs] and had to revert it. The [upstream bugs] are not
+yet fixed.
+
+To work around this for early CI, we will patch
+`machine-config-operator`, `cluster-kube-apiserver-operator`, and
+`cluster-kube-controller-manager-operator` to add the feature gate to
+their operands if and only if the `install-config.yaml` defines a
+dual-stack cluster. (This won't allow migrating from single-stack to
+dual-stack but testing that can wait until after the upstream problems
+are fixed and we can enable the feature gate everywhere.)
+
+[ran into bugs]: https://bugzilla.redhat.com/show_bug.cgi?id=1794376
+[upstream bugs]: https://github.com/kubernetes/kubernetes/pull/86895
+
+#### Upstream Kubernetes
+
+We need to continue watching/shepherding upstream dual-stack KEPs/PRs.
+In theory, not much further work is needed upstream for dual-stack on
+bare metal. (CloudProvider support for IPv6/dual-stack is much
+poorer.)
+
+There are currently very few dual-stack-specific e2e tests upstream
+and we will probably need to create some more.
+
+#### OVN-Kubernetes
+
+Work is ongoing to add dual-stack support to OVN-Kubernetes. Upstream
+OVN-Kubernetes uses CI based on `kind` (Kubernetes-in-Docker) and we
+will have a dual-stack `kind` environment for testing it against
+upstream dual-stack Kubernetes.
+
+#### Installer
+
+The installer already allows configuring dual-stack clusters on
+bare-metal (and Azure) even though they don't yet work, so no work
+should be needed there until/unless we want to support additional
+cloud platforms.
+
+#### Other OCP Components
+
+We believe that *most* components will not need any new work to
+support dual-stack beyond what was done for single-stack IPv6. In
+particular, most services are now listening for both IPv4 and IPv6
+connections regardless of cluster configuration, and all code that
+deals with IP addresses should be capable of dealing with either IPv4
+or IPv6 addresses.
+
+We are currently doing a survey of single-stack-IPv6-related changes
+that were made in the 4.4 cycle to use as a starting point for
+figuring out where further dual-stack changes may be needed. (One
+known problem is with services that only listen on "localhost" rather
+than on all interfaces; there's no trivial way to listen on both
+`127.0.0.1` and `::1`.)
+
+### Risks and Mitigations
+
+#### Completion Risks
+
+If upstream Kubernetes fails to finalize a workable dual-stack
+approach then we would be unable to support dual stack in OCP. We are
+monitoring and contributing to the upstream PRs, KEPs, etc, to make
+sure that this doesn't happen.
+
+#### Functional Risks
+
+We will need to carefully ensure that the `IPv6DualStack` feature gate
+does not change any functionality on single-stack clusters before
+enabling it for all clusters.
```

Comment:
you can note that we have a single-stack Ipv6 CI job that will help mitigate this risk.

Reply:
no, I mean, the worry is that `IPv6DualStack` will break single-stack *IPv4* clusters (as it did when we tried to enable it in 4.4).